### PR TITLE
ModParse fixes

### LINKED ISSUE #2236: Increased Damage with Claws on jewels don't work with Rigwald's Curse on POB
I just checked. The in game dmg calculations work with this mod. So it should work, but is not working in Path of Building.

### PR DESCRIPTION
**Pull request by Nostrademous**
I believe all of this make the `ModParser` more consistent. 

I'm sure this will break some other things though but should make things a lot better in the long run.

This PR will also fix #2236 not needing to have a `KeywordFlag.Hit` specified as it properly changes how:
`["with hits and ailments"]` is treated.

### CODE REVIEW

**Comment by Nostrademous** [comment]
Supersedes #2252 if accepted.

**Comment by pundm** [comment]
I put in your changes but the claw wheel seems broken again. Does it work for you?
image

**Comment by Nostrademous** [comment]
Did you re-generate the cache? CTRL+F5?

Works for me. You should see the ModOrFlag in the Claw Damage.
Blah

**Comment by pundm** [comment]
I did not, sorry. Still trying to get the hang of all this. If this is accepted, can an admin remove my obsolete PR, or do i need to do something on my side?

**Comment by Nostrademous** [comment]
@pundm you should be able to close your own PRs. However, I would leave yours up for now incase my changes here are too much.

**Comment by ALT-F-X** [comment]
This does not fix #2187 even though I thought it would. I thought the problem was with the Ailments parsing which you fixed, is the issue instead because the line doesn't say "Projectile Attacks"?

**Comment by Nostrademous** [comment]
@ALT-F-X what does it think it parsed when you CTRL or ALT hover over the node with the updated ModCache parser?

**Comment by ALT-F-X** [comment]
image

**Comment by Quotae** [comment]
The two Brand mods you changed were done that way intentionally to be able to properly support Arcanist Brand. Arcanist Brand Support adds the Brand SkillType to supported skills, so the mods being done that way lets Brand Damage apply to triggered spells. As far as I know there's no way to add KeywordFlags to supported skills so this was the best way.

**Comment by Nostrademous** [comment]
@Quotae we could redo the `Arcanist Brand` support the same way as I did for CoC/CwC and other trigger abilities. I was thinking about it already, but thanks for clarifying.

**Comment by ALT-F-X** [comment]
@Quotae @Nostrademous Just for my own learning could you explain as to why this wouldn't fix everything with Arcanist Brand? 

image

**Comment by Nostrademous** [comment]
@ALT-F-X b/c that file is auto-generated... so changing it there will get over-written next time an export is done. And the files that generate the export don't have a way to specify additional skillType flags as those are pulled directly from the GGPK tables.

**Comment by ALT-F-X** [comment]
@Nostrademous  You can't do a line like, `#addSkillTypes SkillType.Brand` in the act_int.txt? In looking in the file it looks like someone tried to do something similar but implemented it wrong. Line 1482 of act_int.txt.

**Comment by Nostrademous** [comment]
@ALT-F-X Perhaps. I do not know the entire PoB codebase... probably only a small fraction in reality. I think this is true of many people. What you suggest might be doable and probably would be a clean way to do it but the truth is "I don't know". The only way to find out is to try it and see and then dive-deep with a debugger to see what actually is happening if it doesn't work straight off. 

you would need to implement handing of `#addSkillType` somewhere around here and in a similar fashion:
https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/edc1b96eed7c87630f9bc4ed82fedc34ebaa722f/Export/Scripts/skills.lua#L324

File: Modules/ModParser.lua [hunk 1]
```diff
@@ -2009,6 +2009,14 @@ local specialModList = {
 		mod("Damage", "MORE", tonumber(more) * num / 200, nil, 0, KeywordFlag.Poison, { type = "Condition", var = "DualWielding"}, { type = "SkillType", skillType = SkillType.Attack }),
 		mod("Damage", "MORE", tonumber(more) * num / 100, nil, 0, KeywordFlag.Poison, { type = "Condition", var = "DualWielding", neg = true }, { type = "SkillType", skillType = SkillType.Attack })
 	} end,
+	["enemies poisoned by you have (%-%d+)%% to chaos resistance"] = function(num) return { mod("EnemyModifier", "LIST", { mod = mod("ChaosResist", "BASE", num) }, { type = "ActorCondition", actor = "enemy", var = "Poisoned" }) } end,
```
Comment (Wires77) [inline Modules/ModParser.lua:2012]: This shouldn't be necessary after #2266

**Review by ppoelzl — COMMENTED** [review]
Looks good but we'll have to merge this after automated testing is set up, as these changes could potentially break a lot of builds.

**Pull request closed without merging**